packages/dht/src/dht/discovery/DiscoverySession.ts

```diff
@@ -48,7 +48,7 @@ export class DiscoverySession {
         }
         logger.trace(`Getting closest peers from contact: ${getNodeIdFromPeerDescriptor(contact.getPeerDescriptor())}`)
         this.outgoingClosestPeersRequestsCounter++
-        this.contactedPeers.add(contact.getPeerId().toKey())
+        this.config.contactedPeers.add(contact.getPeerId().toKey())
```

Comment:
In Router we use `setContacted` feature of `SortedContactList`, but maybe the same patterns is not applicable here we have separate lists for each session?

Reply:
The router could be refactored to have its own contacted set similar to the `DiscoverySessions`. 

As the SortedContactList in the RoutingSession is always per session the old setContacted works there as intended.